pkg/tlsx/clients/clients.go

```diff
@@ -400,6 +405,27 @@ func IsZTLSRevoked(options *Options, cert *zx509.Certificate) bool {
 	return IsTLSRevoked(options, xcert)
 }
 
+// IsUntrustedCA returns true if the certificate is a self-signed CA
+func IsUntrustedCA(certs []*x509.Certificate) bool {
+	for _, c := range certs {
+		if c.IsCA && IsSelfSigned(c.AuthorityKeyId, c.SubjectKeyId) {
+			return true
+		}
+	}
+	return false
+}
+
+// IsZTLSUntrustedCA returns true if the certificate is a self-signed CA
+func IsZTLSUntrustedCA(certs []tls.SimpleCertificate) bool {
+	for _, cert := range certs {
+		parsedCert, _ := x509.ParseCertificate(cert.Raw)
+		if parsedCert.IsCA && IsSelfSigned(parsedCert.AuthorityKeyId, parsedCert.SubjectKeyId) {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
@ShubhamRasal , I think we can add nil check here to see if cert is nil i.e `parsedCert != nil` , `c != nil` to avoid nil pointer dereference . Since we are parsing certificates in openssl we can't be too sure

Reply:
Added pointer nil check condition